src/feedlistformaction.cpp

```diff
@@ -150,11 +150,20 @@ void FeedListFormAction::process_operation(Operation op,
 			  "(f)irsttag/(t)itle/(a)rticlecount/"
 			  "(u)nreadarticlecount/(l)astupdated/(n)one?"),
 			input_options);
-		if (!c)
+		if (!c) {
 			break;
-		unsigned int n_options = ((std::string) "ftaun").length();
-		if (input_options.length() < n_options)
+		}
+
+		// Check that the number of translated answers is the same as the
+		// number of answers we expect to handle. If it doesn't, just give up.
+		// That'll prevent this function from sorting anything, so users will
+		// complain, and we'll ask them to update the translation. A bit lame,
+		// but it's better than mishandling the answer.
+		const auto n_options = ((std::string) "ftaun").length();
+		if (input_options.length() < n_options) {
 			break;
+		}
```

Comment:
I don't think so. We can't `break` from  from inside a function, so we'll have to check function's return value and `break` conditionally—which is no different from the `if` we already have.

These four lines could be packed into a C macro, but implicit dependency on `input_options` would make that a bit brittle. And we can't abstract away the repetition of `"ftaun"`, because the one marked with `_` (which expands to `gettext`) should be present in code verbatim in order to be picked up by `xgettext`.

I decided to call this a "pattern", bite the bullet, and repeat it throughout the codebase.